### PR TITLE
Update guzzle client

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 - Client factories for Buzz.
 
+### Changed
+
+- Guzzle 6 client now uses the new `createWithConfig` named constructor
```
Comment (dbu) [inline CHANGELOG.md]: this is very technical. i think for the user, its more relevant to know what that means:

- Guzzle 6 client is now created according to the Httplug specifications with automated minimal behaviour. Make sure you configure the Httplug plugins as needed, for example if you want to get exceptions for failure HTTP status codes.

**Comment by sagikazarmark** [comment]
@dbu anything else here from your POV?

File: composer.json [hunk 2]
```diff
@@ -37,6 +37,7 @@
         "puli/symfony-bundle": "^1.0@beta"
     },
     "conflict": {
+        "php-http/guzzle6-adapter": "<1.1",
```
Comment (Nyholm) [inline composer.json]: Im not sure this line is needed.

**Comment by Nyholm** [comment]
👍

File: composer.json (hunk 2 above)
Comment (sagikazarmark) [inline composer.json]: If someone installs 1.0.0, Guzzle 6 factory will emit an error.

**Comment by dbu** [comment]
thanks!

**Pull request merged**